.github/workflows/r_package.yml

```diff
@@ -276,53 +287,21 @@ jobs:
         with:
           fetch-depth: 5
           submodules: true
-      - name: update to clang 15
-        shell: bash
+      - name: install clang
         run: |
-          # remove clang stuff that comes installed in the image
-          apt-get autoremove -y --purge \
-              clang-* \
-              libclang-* \
-              libunwind-* \
-              llvm-*
-          #
-          # replace it all with clang-15
-          apt-get update -y
-          apt-get install --no-install-recommends -y \
-              gnupg \
-              lsb-release \
-              software-properties-common \
-              wget
-          #
-          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-          #
-          add-apt-repository "deb http://apt.llvm.org/unstable/ llvm-toolchain main"
-          apt-get install -y --no-install-recommends \
-              clang-15 \
-              clangd-15 \
-              clang-format-15 \
-              clang-tidy-15 \
-              clang-tools-15 \
-              lldb-15 \
-              lld-15 \
-              llvm-15-dev \
-              llvm-15-tools \
-              libomp-15-dev \
-              libc++-15-dev \
-              libc++abi-15-dev \
-              libclang-common-15-dev \
-              libclang-15-dev \
-              libclang-cpp15-dev \
-              libunwind-15-dev
-          # overwrite everything in /usr/bin with the new v15 versions
-          cp --remove-destination /usr/lib/llvm-15/bin/* /usr/bin/
+          ./.ci/install-clang-devel.sh ${{ matrix.clang-version }}
       - name: Install packages and run tests
         shell: bash
         run: |
           export PATH=/opt/R-devel/bin/:${PATH}
           Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'markdown', 'Matrix', 'RhpcBLASctl', 'testthat'), repos = 'https://cran.rstudio.com', Ncpus = parallel::detectCores())"
           sh build-cran-package.sh
           R CMD check --as-cran --run-donttest lightgbm_*.tar.gz || exit -1
+          echo ""
+          echo "install logs:"
+          echo ""
+          cat lightgbm.Rcheck/00install.out
+          echo ""
```

Comment:
Without this, compiler invocations and most warnings are not visible in CI logs because `R CMD check` redirects them to this file instead of printing them.

I think this will be generically useful for CI in the future, and in fact we already do that for most other R CI jobs:

https://github.com/microsoft/LightGBM/blob/2e3543c04c4b8e475966721ad8570544edd5b470/.ci/test_r_package.sh#L226-L232